pkg/kclient/ingress.go

```diff
@@ -22,29 +48,120 @@ func (c *Client) CreateIngress(ingress extensionsv1.Ingress) (*extensionsv1.Ingr
 	return ingressObj, nil
 }
 
-// DeleteIngress deletes the given ingress
 func (c *Client) DeleteIngress(name string) error {
+	var errList []error
+	var err error
+	if c.isNetworkingV1IngressSupported {
+		err = c.KubeClient.NetworkingV1().Ingresses(c.Namespace).Delete(context.TODO(), name, metav1.DeleteOptions{})
+		if err != nil {
+			errList = append(errList, fmt.Errorf("failed to delete networking v1 beta 1 ingress %w", err))
+		} else {
+			return nil
+		}
+	}
+	if c.isExtensionV1Beta1IngressSupported {
+		err = c.KubeClient.ExtensionsV1beta1().Ingresses(c.Namespace).Delete(context.TODO(), name, metav1.DeleteOptions{})
+		if err != nil {
+			errList = append(errList, fmt.Errorf("failed to delete extension v1beta1 ingress %w", err))
+		} else {
+			return nil
+		}
+	}
+	reterr := fmt.Errorf("failed to delete ingress due to reasons")
+	for _, item := range errList {
+		reterr = fmt.Errorf("%s %w", reterr.Error(), item)
+	}
+	return reterr
+}
+
+// DeleteExtensionV1Beta1Ingress deletes the given ingress
+func (c *Client) DeleteExtensionV1Beta1Ingress(name string) error {
 	err := c.KubeClient.ExtensionsV1beta1().Ingresses(c.Namespace).Delete(context.TODO(), name, metav1.DeleteOptions{})
 	if err != nil {
 		return errors.Wrap(err, "unable to delete ingress")
 	}
 	return nil
 }
 
-// ListIngresses lists all the ingresses based on the given label selector
-func (c *Client) ListIngresses(labelSelector string) ([]extensionsv1.Ingress, error) {
-	ingressList, err := c.KubeClient.ExtensionsV1beta1().Ingresses(c.Namespace).List(context.TODO(), metav1.ListOptions{
-		LabelSelector: labelSelector,
-	})
-	if err != nil {
-		return nil, errors.Wrap(err, "unable to get ingress list")
+func (c *Client) ListIngresses(labelSelector string) ([]*unions.KubernetesIngress, error) {
+	var kil []*unions.KubernetesIngress
+	var errList []error
+	if c.isNetworkingV1IngressSupported {
+		ingressList, err1 := c.KubeClient.NetworkingV1().Ingresses(c.Namespace).List(context.TODO(), metav1.ListOptions{
```

Comment:
There is no `err`, you can use `err` here

Reply:
updated